### PR TITLE
#7 - Move special behaviour from factories to Incremental DOM

### PR DESCRIPTION
**Pull request by treshugart**
Implements #7.

### CODE REVIEW

File: src/vdom.js [hunk 1]
```diff
@@ -32,39 +32,56 @@ attributes.key = attributes.skip = attributes.statics = function () {};
...
+    return applyProp(elem, name, value);
+  }
+
+  // Handle custom events.
+  if (name.indexOf('on') === 0) {
+    return applyEvent(elem, name.substring(2), name, value);
+  }
+
+  // Custom element properties should be set as properties.
+  const dataName = elem.tagName + '.' + name;
```
Comment (treshugart) [inline src/vdom.js:53]: I wonder if we need this if we're doing the `name in elem` above?

File: src/vdom.js [hunk 2]
```diff
@@ -32,39 +32,56 @@ attributes.key = attributes.skip = attributes.statics = function () {};
...
   if (internalData.applyProp[dataName]) {
```
Comment (jpnelson) [inline src/vdom.js:54]: When does `internalData.applyProp[dataName]` get set?

**Comment by jpnelson** [comment]
lgtm

**Comment by bqx** [comment]
unrelated, but in https://github.com/skatejs/kickflip/pull/8/files#diff-016f7233da52885423a5118717cf4e88R4 the word 'so' is repeated twice

File: src/vdom.js [hunk 3]
```diff
@@ -32,39 +32,56 @@ attributes.key = attributes.skip = attributes.statics = function () {};
...
+
+  // Fallback to default IncrementalDOM behaviour.
+  applyDefault(elem, name, value);
 };
 
-function applyEvent (eName) {
-  return function (elem, name, value) {
-    let events = elem.__events;
+// Adds or removes an event listener for an element.
+function applyEvent (elem, ename, name, value) {
```
Comment (bqx) [inline src/vdom.js:63]: `name` seems to be unused in this function.

**Comment by treshugart** [comment]
Thanks @bqx, I fixed that in 49bc55244b70ca09b14e2ff92bc4c73a56866bc1.

File: src/vdom.js [hunk 4]
```diff
@@ -32,39 +32,56 @@ attributes.key = attributes.skip = attributes.statics = function () {};
...
+  let events = elem.__events;
 
-    if (!events) {
-      events = elem.__events = {};
-    }
+  if (!events) {
+    events = elem.__events = {};
+  }
```
Comment (bqx) [inline src/vdom.js:68]: I think this would be easier to read:

```
elem.__events = elem.__events || {};
const events = elem.__events;
```

**Comment by bqx** [comment]
lgtm

**Pull request merged**